framework/src/outputs/OutputWarehouse.C

```diff
@@ -178,6 +180,19 @@ OutputWarehouse::mooseConsole()
     _console_buffer.clear();
     _console_buffer.str("");
   }
+  else
+  {
+    if (!_buffer_action_console_outputs)
+    {
+      // this will cause messages to console before its construction immediately flushed and cleared.
+      std::string message = _console_buffer.str();
+      if (_app.multiAppLevel() > 0)
+        MooseUtils::indentMessage(_app.name(), message);
+      Moose::out << message;
+      _console_buffer.clear();
+      _console_buffer.str("");
```

Comment:
See the other branch. I am copying these two lines. Not sure if it is necessary.